Sources/Sentry/SentryClient.m

```diff
@@ -808,9 +809,40 @@ - (void)applyExtraDeviceContextToEvent:(SentryEvent *)event
                     key:@"app"
                   block:^(NSMutableDictionary *app) {
                       [app addEntriesFromDictionary:extraContext[@"app"]];
+#if SENTRY_HAS_UIKIT
+                      [self addViewNamesToContext:app event:event];
+#endif // SENTRY_HAS_UIKIT
                   }];
 }
 
+#if SENTRY_HAS_UIKIT
+- (void)addViewNamesToContext:(NSMutableDictionary *)appContext event:(SentryEvent *)event
+{
+    if ([event isKindOfClass:[SentryTransaction class]]) {
+        SentryTransaction *transaction = (SentryTransaction *)event;
+        if ([transaction.screens count] > 0) {
+            appContext[@"view_names"] = transaction.screens;
+        }
```

Comment:
`h`: If I'm not mistaken, we must only attach `view_names` for transactions when they have one. If not, we should exit here.
```suggestion
        }
        return;
```

Reply:
Well, this feature was defined by us. Do you think this is the best approach?